### PR TITLE
[SPARK-14645][MESOS] Fix python running on cluster mode mesos to have non local uris

### PR DESCRIPTION
**Pull request by tnachen**
## What changes were proposed in this pull request?

Fix SparkSubmit to allow non-local python uris
## How was this patch tested?

Manually tested with mesos-spark-dispatcher

### CODE REVIEW

**Comment by tnachen** [comment]
@dragos @andrewor14 PTAL, it's a very small change.

File: core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala [hunk 1]
```diff
@@ -306,7 +306,7 @@ object SparkSubmit {
 
     // Require all python files to be local, so we can add them to the PYTHONPATH
     // In YARN cluster mode, python files are distributed as regular files, which can be non-local
```
Comment (andrewor14) [inline core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala]: need to update this comment?

**Comment by andrewor14** [comment]
@tnachen there are a few places where `isYarnCluster` is used. Can you check if we need to do it for those places as well? (e.g. R)

**Comment by tnachen** [comment]
@andrewor14 fixed the comment.
I just checked the other places where isYarnCluster is used, and looks like they're still correct. Spark on Mesos doesn't yet support R (at least not really tested). I'll be taking a look at that next.

**Comment by dragos** [comment]
LGTM

**Comment by tnachen** [comment]
@andrewor14 PTAL

**Comment by andrewor14** [comment]
Thanks sorry for the wait but LGTM merging into master 2.0

**Pull request closed without merging**